src/bubble_swarm/utils.rs

```diff
@@ -0,0 +1,75 @@
+use crate::math2d::{circle::Circle, point::Point, regioned_angle::RegionedAngle, vector::Vector};
+
+use super::{angle_area::AngleArea, source_circle::SourceCircle};
+
+pub fn get_mid_point(from: Point, to: Point) -> Point {
+    from + Vector::from((from, to)).scale(0.5)
+}
+
+pub enum CircleIntersection {
```

Comment:
so cool that you use rust enums here :)

Reply:
I also want to learn those more :) My idea was that in the is_within_angle_of_source_circle function (and wherever else this function will be called), we won't need to do three separate if checks, but could do some nice pattern matching :)